### DISCUSSION TITLE
Plug-ins inside a function

### DISCUSSION THREAD

**Discussion by jasonlim131** [jspsych/jsPsych#760] [Q&A]
Hello github I'm new to JsPsych, and I want to create a face images encoding session. I organized a test block into a generateTestBlock function with sub-functions that take the same input: 
```
function generateTestBlock(list) {
    start = generateListStart(list)
    encode = displayList(list)
    distractor = generateDistractor(list) //done
    recognition = recognitionTest(list)
    return [start, encode, distractor, recognition]
}
```
To display array of faces, I use jspsych-animation.
I understand that in order to run a plug-in, i have to place the plug-in variable inside the timeline array. If the plugin is inside the function displayList(list), will the plug-in play when the function is called and return its values? Thanks!

**Comment by vijaymarupudi** [comment]
I'm not sure I understand your question, are you asking the plugin to be called inside the displayList() function and for that function to return after it is displayed? If so, I don't believe the design of jsPsych makes this possible.

**Comment by jodeleeuw** [comment]
I think we need more information here about how you are constructing the timeline from the generateTestBlock function. Is the array that you are returning a set of trials? If so, you could `concat` the array onto a timeline array.

**Comment by jkhartshorne** [comment]
@jasonlim131 any updates?
